File: iosApp/iosApp/Login/LoginState.swift
import Foundation

/// Immutable snapshot of everything the Login screen needs to render itself.
///
/// A new value is published by `LoginViewModel` for every state change;
/// the View never mutates this struct directly.
struct LoginState {

    /// Current text in the email input field.
    var email: String = ""

    /// Current text in the password input field.
    var password: String = ""

    /// When `true` the password field renders plain text instead of dots.
    var isPasswordVisible: Bool = false

    /// `true` while the login request is in-flight; drives the spinner
    /// and disables the submit button.
    var isLoading: Bool = false

    /// Non-`nil` when a validation or server error should be shown to the user.
    /// Cleared by sending `LoginIntent.dismissError`.
    var errorMessage: String? = nil
}

File: composeApp/src/androidMain/kotlin/org/gce/racehub/login/LoginState.kt
package org.gce.racehub.login

/**
 * Immutable snapshot of everything the Login screen needs to render itself.
 *
 * A new instance is produced by the ViewModel for every state change;
 * the View never mutates this object directly.
 */
data class LoginState(

    /** Current text in the email input field. */
    val email: String = "",

    /** Current text in the password input field. */
    val password: String = "",

    /** When true the password field renders plain text instead of dots. */
    val isPasswordVisible: Boolean = false,

    /** True while the login request is in-flight; used to show a spinner
     *  and disable the submit button. */
    val isLoading: Boolean = false,

    /** Non-null when a validation or server error should be shown to the user.
     *  Cleared by dispatching [LoginIntent.DismissError]. */
    val errorMessage: String? = null
)
